### PR TITLE
Skyscrapers puzzle editor

### PR DESCRIPTION
**Pull request by jadeandtea**
## Description

Implemented a puzzle editor for the Skyscraper puzzle editor. I left in the ability to edit tile numbers, but those values will not be saved with the puzzle.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Enhancement (improvement to an already existing feature)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

I was able to create a new skyscraper puzzle, edit the tiles and clue values, and save the file. I was able to reopen the new puzzle in both the puzzle solver and the edit existing puzzle menus.

## Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by charlestian23** [comment]
@jadeandtea Can you resolve the merge conflict?

File: src/main/java/edu/rpi/legup/puzzle/skyscrapers/SkyscrapersExporter.java
Comment (charlestian23) [inline src/main/java/edu/rpi/legup/puzzle/skyscrapers/SkyscrapersExporter.java:1]: I'm getting an error on this file when I try to build the jar file. I think there's some instances of `board.getLines()` that haven't been safely removed.

**Review by charlestian23 — CHANGES_REQUESTED** [review]
Most of the files seem okay, made a comment on the file that had an issue.

File: src/main/java/edu/rpi/legup/puzzle/skyscrapers/SkyscrapersExporter.java
Comment (jadeandtea) [reply src/main/java/edu/rpi/legup/puzzle/skyscrapers/SkyscrapersExporter.java:1]: I think I fixed it?

**Comment by charlestian23** [comment]
It seems like a lot of the tests for Skyscrapers are failing. I'll resolve the [above thread](https://github.com/Bram-Hub/LEGUP/pull/720#discussion_r1480650746
) once those issues are resolved.

**Review by charlestian23 — APPROVED** [review]
Looks fine

**Pull request merged**